fem/fespace.hpp

```diff
@@ -660,76 +781,261 @@ class FiniteElementSpace
+   ///@{
 
-   /** @brief Returns the indices of all of the VDofs for the specified
-       dimension 'vd'. */
-   /** The 'ndofs' parameter defines the number of Dofs in the
-       FiniteElementSpace. If 'ndofs' is -1 (the default value), then the
-       number of Dofs is determined by the FiniteElementSpace. */
+   /// @brief Returns the indices of all of the VDofs for the specified
+   /// dimension 'vd'.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
+   ///
+   /// @note This method does not resize the @a dofs array. It takes the range
+   /// of dofs [0, dofs.Size()) and converts these to @ref vdof "vdofs" and
+   /// stores the results in the @a dofs array.
    void GetVDofs(int vd, Array<int> &dofs, int ndofs = -1) const;
 
+   /// @brief Compute the full set of @ref vdof "vdofs" corresponding to each
+   /// entry in @a dofs.
+   ///
+   /// @details Produces a set of @ref vdof "vdofs" of
+   /// length @b vdim * dofs.Size() corresponding to the entries contained in
+   /// the @a dofs array.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
+   ///
+   /// @note The @a dofs array is overwritten and resized to accomodate the
+   /// new values.
    void DofsToVDofs(Array<int> &dofs, int ndofs = -1) const;
 
+   /// @brief Compute the set of @ref vdof "vdofs" corresponding to each entry
+   /// in @a dofs for the given vector index @a vd.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
+   ///
+   /// @note The @a dofs array is overwritten with the new values but its size
+   /// will not be altered.
    void DofsToVDofs(int vd, Array<int> &dofs, int ndofs = -1) const;
 
+   /// @brief Compute a single @ref vdof corresponding to the index @a dof and
+   /// the vector index @a vd.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
    int DofToVDof(int dof, int vd, int ndofs = -1) const;
 
+   /// @brief Compute the inverse of the Dof to VDof mapping for a single
+   /// index @a vdof.
+   ///
+   /// @warning This method is only intended for use with positive indices.
+   /// Passing a negative value for @a vdof will produce an invalid result.
```

Comment:
Do we want to put an MFEM_ASSERT here to catch negative values?

Reply:
Probably a wise idea. If the @tzanio and @jamiebramwell are ok with making a functional change in this PR I'll add it. Otherwise we can do this in another PR.